plugins/plugin-chart-echarts/src/Gauge/controlPanel.tsx

```diff
@@ -0,0 +1,284 @@
+        ],
+        [
+          {
+            name: 'show_split_line',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Show split lines'),
+              description: t('Whether to show the split lines on the axis'),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.showSplitLine,
+            },
+          },
+        ],
+        [
+          {
+            name: 'split_number',
+            config: {
+              type: 'SliderControl',
+              label: t('Split number'),
+              description: t('Number of split segments on the axis'),
+              renderTrigger: true,
+              min: 3,
+              max: 30,
+              default: DEFAULT_FORM_DATA.splitNumber,
+            },
+          },
+        ],
+        [<h1 className="section-header">{t('Progress')}</h1>],
+        [
+          {
+            name: 'show_progress',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Show progress'),
+              description: t('Whether to show the progress of gauge chart'),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.showProgress,
+            },
+          },
+        ],
+        [
+          {
+            name: 'overlap',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Overlap'),
+              description: t(
+                'Whether the progress bar overlaps when there are multiple groups of data',
+              ),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.overlap,
+            },
+          },
+        ],
+        [
+          {
+            name: 'roundcap',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Roundcap'),
+              description: t('Style the ends of the progress bar with a roundcap'),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.roundcap,
+            },
+          },
+        ],
+        [<h1 className="section-header">{t('Intervals')}</h1>],
+        [
+          {
+            name: 'intervals',
+            config: {
+              type: 'TextControl',
+              label: t('Interval bounds'),
+              description: t(
+                'Comma-separated interval bounds, e.g. 2,4,5 for intervals 0-2, 2-4 and 4-5. Last number should match the value provided for MAX.',
+              ),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.intervals,
+            },
+          },
+        ],
+        [
+          {
+            name: 'interval_color_indices',
+            config: {
+              type: 'TextControl',
+              label: t('Interval colors'),
+              description: t(
+                'Comma-separated color picks for the intervals, e.g. 1,2,4. Integers denote colors from the chosen color scheme and are 1-indexed. Length must be matching that of interval bounds.',
```

Comment:
Should we add a validator here to take care of these restrictions or handle those explicitly?
like, omit 0 if present,if the last number is not max then override it in transformprops.

Reply:
sounds useful but I don't see a way to fetch full form data in the validator... I would need to check the last number against maxVal, and validator only gets the field it's validating but not the rest

Reply:
@mayurnewase do you think dropping this would be ok?

Reply:
yeah, I guess other than this no other comments from me.

Reply:
I think we could take the modulo on the the length of the color scheme when using it, thereby looping and avoiding errors.

Reply:
Please see here:
https://github.com/apache-superset/superset-ui/pull/993/commits/07763d34cb388130ad462061bdf689dbff7f1213
I can squash all commits before merging if needed.

Reply:
Thanks @krsnik93 - my bad! No need to squash, as me always squash merge PRs. Looking forward to getting this merged soon!